### ISSUE TITLE
[declarative-hack] Adding conversationProperty to WelcomeRule fires WelcomeRule twice

### ISSUE THREAD

**Issue by Zerryth** [microsoft/botbuilder-js#830]
## Versions
In stevenic/4.4-planning branch for hackathon, using declarative node version

## Describe the bug
Running sample `04.welcomeRule`, this is what's checked in for the user greeting

```js
// Greet User
dialogs.addRule(new WelcomeRule([
    new SendActivity(`Welcome! Say "hello" to get started.`)
], `user.greeted`));
```

However the greeting is fired twice. Seems like once when the conversation starts, and another when the user starts the dialog

![image](https://user-images.githubusercontent.com/35248895/54849195-cc71d000-4ca0-11e9-9ebb-742d6376ecd6.png)

Deleting `user.greeted` allows for the WelcomeRule to fire only once as expected in sample 04

## Expected behavior
For WelcomeRule to trigger only if a new plan is started or a new user is recognized. 

[bug]

**Comment by Stevenic** [comment]
Old issue
